### PR TITLE
Support collapsing multi-line comments in any context

### PR DESCRIPTION
**Pull request by dibarbet**
Fixes https://github.com/dotnet/roslyn/issues/64001

Looks like
image

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/Structure/AccessorDeclarationStructureTests.cs [hunk 1]
```diff
@@ -58,28 +58,6 @@ public string Text
                 Region("textspan2", "hint2", CSharpStructureHelpers.Ellipsis, autoCollapse: true));
         }
 
-        [Fact]
-        public async Task TestPropertyGetterWithMultiLineComments1()
```
Comment (dibarbet) [inline src/EditorFeatures/CSharpTest/Structure/AccessorDeclarationStructureTests.cs:62]: tests here were moved

File: src/EditorFeatures/CSharpTest/Structure/CommentStructureTests.cs [hunk 2]
```diff
@@ -104,56 +104,5 @@ class C
             await VerifyBlockSpansAsync(code,
                 Region("span", "// Hello ...", autoCollapse: true));
         }
-
-        [Fact]
-        public async Task TestMultilineComment1()
```
Comment (dibarbet) [inline src/EditorFeatures/CSharpTest/Structure/CommentStructureTests.cs:109]: tests here were moved to the new test file

File: src/EditorFeatures/CSharpTest/Structure/CompilationUnitStructureTests.cs [hunk 3]
```diff
@@ -150,15 +150,6 @@ public async Task TestUsingKeywordWithSpace()
                 Region("textspan", "hint", CSharpStructureHelpers.Ellipsis, autoCollapse: true));
         }
 
-        [Fact, WorkItem(16186, "https://github.com/dotnet/roslyn/issues/16186")]
-        public async Task TestInvalidComment()
```
Comment (dibarbet) [inline src/EditorFeatures/CSharpTest/Structure/CompilationUnitStructureTests.cs:154]: moved

File: src/Features/CSharp/Portable/Structure/CSharpStructureHelpers.cs [hunk 4]
```diff
@@ -138,7 +138,7 @@ private static string CreateCommentBannerTextWithPrefix(string text, string pref
             return prefix + " " + text.Substring(prefixLength).Trim() + " " + Ellipsis;
         }
 
-        private static string GetCommentBannerText(SyntaxTrivia comment)
+        internal static string GetCommentBannerText(SyntaxTrivia comment)
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Structure/CSharpStructureHelpers.cs:141]: consider just making public.

**Review by CyrusNajmabadi — APPROVED** [review]

**Pull request merged**